### PR TITLE
Capsule update command for SVN commit

### PR DESCRIPTION
**Pull request by lsubash**
Capsule Command support added for anti rollback
security version number. User can create command
in text file and create capsule with CMDI mode.

{ARBSVNCOMMIT}

Platform APIs would be invoked to do SVN
commit operations by useing HECI interfaces.

Signed-off-by: Subash Lakkimsetti <subash.lakkimsetti@intel.com>

### CODE REVIEW

File: BootloaderCommonPkg/Include/Library/FirmwareUpdateLib.h [hunk 1]
```diff
@@ -526,4 +526,21 @@ SetFlashDescriptorLock (
...
    IN  UINTN     CmdDataSize
    );
+
+/**
+  Anti Rollback Svn Commit
+
+  @param[in]  CmdDataBuf    Pointer to command buffer.
+  @param[in]  CmdDataSize   size of command data.
+
+  @retval  EFI_SUCCESS      Flash descriptor lock successfully.
```
Comment (sagraw2) [inline BootloaderCommonPkg/Include/Library/FirmwareUpdateLib.h:536]: Fix the comment.

Comment (lsubash) [reply BootloaderCommonPkg/Include/Library/FirmwareUpdateLib.h]: Thanks Sachin. Updated PR.

File: PayloadPkg/FirmwareUpdate/CmdFwUpdate.c [hunk 2]
```diff
@@ -51,6 +52,9 @@ FwUpdateCmdHandler (
   if(AsciiStrnCmp(CmdBuf, FLASH_DESCRIPTOR_LOCK_STR, AsciiStrLen(FLASH_DESCRIPTOR_LOCK_STR)) == 0) {
```
Comment (mauricema) [inline PayloadPkg/FirmwareUpdate/CmdFwUpdate.c:52]: put a space after "if"

File: PayloadPkg/FirmwareUpdate/CmdFwUpdate.c [hunk 3]
```diff
@@ -51,6 +52,9 @@ FwUpdateCmdHandler (
   if(AsciiStrnCmp(CmdBuf, FLASH_DESCRIPTOR_LOCK_STR, AsciiStrLen(FLASH_DESCRIPTOR_LOCK_STR)) == 0) {
       *CmdProcessed  =  CMDI_TYPE_SPI_DESCRIPTOR_LOCK;
       Status = SetFlashDescriptorLock (CmdBuf, BufLen);
+  } else if(AsciiStrnCmp(CmdBuf, ARB_SVN_COMMIT_STR, AsciiStrLen(ARB_SVN_COMMIT_STR)) == 0) {
```
Comment (mauricema) [inline PayloadPkg/FirmwareUpdate/CmdFwUpdate.c:55]: put a space after "if"

Comment (mauricema) [inline PayloadPkg/FirmwareUpdate/CmdFwUpdate.c:55]: why not use AsciiStrCmp () ?

Comment (lsubash) [reply PayloadPkg/FirmwareUpdate/CmdFwUpdate.c:55]: Currently we recommend command format as " {ARBSVNCOMMIT}
We would also need to support " {ARBSVNCOMMIT param1 param2..}
In this case we need to check for keyword using AsciiStrnCmp() to find keyword is present or not in string of words.

**Comment by lsubash** [comment]
Updated PR with comments fixed.

File: PayloadPkg/FirmwareUpdate/CmdFwUpdate.c [hunk 4]
```diff
@@ -48,9 +49,12 @@ FwUpdateCmdHandler (
   Status = EFI_SUCCESS;
   *CmdProcessed = 0;
 
-  if(AsciiStrnCmp(CmdBuf, FLASH_DESCRIPTOR_LOCK_STR, AsciiStrLen(FLASH_DESCRIPTOR_LOCK_STR)) == 0) {
+  if(AsciiStrnCmp (CmdBuf, FLASH_DESCRIPTOR_LOCK_STR, AsciiStrLen(FLASH_DESCRIPTOR_LOCK_STR)) == 0) {
```
Comment (myguitar) [inline PayloadPkg/FirmwareUpdate/CmdFwUpdate.c:52]: A space between if and (.
```
if(AsciiStrnCmp
->
if (AsciiStrnCmp
```

File: PayloadPkg/FirmwareUpdate/CmdFwUpdate.c [hunk 5]
```diff
@@ -48,9 +49,12 @@ FwUpdateCmdHandler (
   Status = EFI_SUCCESS;
   *CmdProcessed = 0;
 
-  if(AsciiStrnCmp(CmdBuf, FLASH_DESCRIPTOR_LOCK_STR, AsciiStrLen(FLASH_DESCRIPTOR_LOCK_STR)) == 0) {
+  if(AsciiStrnCmp (CmdBuf, FLASH_DESCRIPTOR_LOCK_STR, AsciiStrLen(FLASH_DESCRIPTOR_LOCK_STR)) == 0) {
       *CmdProcessed  =  CMDI_TYPE_SPI_DESCRIPTOR_LOCK;
       Status = SetFlashDescriptorLock (CmdBuf, BufLen);
+  } else if(AsciiStrnCmp (CmdBuf, ARB_SVN_COMMIT_STR, AsciiStrLen(ARB_SVN_COMMIT_STR)) == 0) {
```
Comment (myguitar) [inline PayloadPkg/FirmwareUpdate/CmdFwUpdate.c:55]: A space between if and (.

File: PayloadPkg/FirmwareUpdate/CmdFwUpdate.c (hunk 3 above)
Comment (mauricema) [reply PayloadPkg/FirmwareUpdate/CmdFwUpdate.c]: got it.

**Pull request merged**